clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2708,6 +2708,14 @@ void CodeGenModule::EmitDeferred() {
   CurDeclsToEmit.swap(DeferredDeclsToEmit);
 
   for (GlobalDecl &D : CurDeclsToEmit) {
+    if (LangOpts.CUDA && LangOpts.SYCLIsHost) {
```

Comment:
Comment would be very helpful. Please add a comment.

Reply:
Yes, good point! Thank you, @premanandrao.

I added a commit in this regard.